### PR TITLE
CI: extract common logic into setup action

### PR DESCRIPTION
**Pull request by athre0z**
Based on https://github.com/elastic/otel-profiling-agent/pull/42 and waiting for that to merge to undraft.

### CODE REVIEW

**Review by florianl — APPROVED** [review]

File: .github/workflows/unit-test-on-pull-request.yml [hunk 1]
```diff
@@ -16,37 +16,15 @@ jobs:
...
-      id: go
-    - name: Check out
-      uses: actions/checkout@v4
-    - name: Linter
-      run: |
-        go version
-        curl -sSfL https://raw.githubusercontent.com/golangci/golangci-lint/master/install.sh | sh -s -- -b $(go env GOPATH)/bin v1.56.2
-        make lint
+      - name: Clone code
+        uses: actions/checkout@v4
```
Comment (rockdaboot) [inline .github/workflows/unit-test-on-pull-request.yml:20]: We have this repeated at 3 places. I'd consider this as part of setting up the environment.

**Review by rockdaboot — APPROVED** [review]
Left a non-blocking comment

File: .github/workflows/unit-test-on-pull-request.yml (hunk 1 above)
Comment (florianl) [reply .github/workflows/unit-test-on-pull-request.yml:20]: In the setup environment step, a file `.github/workflows/env`, is referenced that is only available if the code is checked out before as it is part of the repository.

Comment (rockdaboot) [reply .github/workflows/unit-test-on-pull-request.yml:20]: Ah, that makes sense :)

**Pull request merged**